### PR TITLE
Feature Flags: Remove dev-mode and alpha from grpc server feature flag

### PR DESCRIPTION
**Pull request by gotjosh**
The Alerting squad is about to use the grpc server for a production-grade feature, let's make sure we remove:

- The dev mode only flag.
- The alpha tag. Let's promote it to beta.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -425,6 +425,8 @@ steps:
     CGO_ENABLED: "1"
   image: golang:1.20.1
   name: lint-backend
+- commands:
+    - make proto-lint
```
Comment (DanCech) [inline .drone.yml:429]: what is this for?

Comment (DanCech) [reply .drone.yml:429]: if we need to update drone.yml the change needs to be implemented in the .star files that we use to generate drone.yml

**Review by toddtreece — COMMENTED** [review]
the change to the feature toggle looks good 👍

File: .drone.yml (hunk 1 above)
Comment (gotjosh) [reply .drone.yml]: Sorry, completely unrelated commit.

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ Some stable features are enabled by default. You can disable a stable feature by
 | `validateDashboardsOnSave`                | Validate dashboard JSON POSTed to api/dashboards/db                                                                                                                                          |
 | `autoMigrateOldPanels`                    | Migrate old angular panels to supported versions (graph, table-old, worldmap, etc)                                                                                                           |
 | `disableAngular`                          | Dynamic flag to disable angular at runtime. The preferred method is to set `angular_support_enabled` to `false` in the [security] settings, which allows you to change the state at runtime. |
+| `grpcServer`                              | Run GRPC server                                                                                                                                                                              |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:48]:
```suggestion
| `grpcServer`                              | Runs the GRPC server                                                                                                                                                                              |
```

**Review by chri2547 — COMMENTED** [review]
A small docs suggestion. Thank you for the contribution!

**Review by ryantxu — APPROVED** [review]
LGTM 🌮

**Pull request merged**